From 12fed6178ad5ada71d591ec1f948e7df30f87db8 Mon Sep 17 00:00:00 2001
From: William Jeffreys <wjeffreys96@gmail.com>
Date: Sun, 10 May 2026 18:50:31 -0700
Subject: [PATCH] scripts: runners: fix probe-rs gdb not handling SIGINT

As in PR #106778, using check_call to start up the gdb server doesn't
ignore SIGINT meaning it will exit the debug session without cleaning
up. Using check_call_ignore_sigint fixes the issue.

Signed-off-by: William Jeffreys <wjeffreys96@gmail.com>
---
 scripts/west_commands/runners/probe_rs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/west_commands/runners/probe_rs.py b/scripts/west_commands/runners/probe_rs.py
index 23844bcd6c308..909f1e2845422 100644
--- a/scripts/west_commands/runners/probe_rs.py
+++ b/scripts/west_commands/runners/probe_rs.py
@@ -177,7 +177,7 @@ def do_debug_debugserver(self, command, **kwargs):
             # debugserver mode
             self.logger.info(f'probe-rs GDB server running on port {self.gdb_port}')
 
-        self.check_call([self.probe_rs, 'gdb']
+        self.check_call_ignore_sigint([self.probe_rs, 'gdb']
                         + self.args + debug_args)
 
     def do_attach(self, **kwargs):
